src/store.ts

```diff
@@ -11,35 +11,62 @@ interface ICycle {
 }
 
 const startingCycles: ICycle[] = [
-  {cursor: 2, length: 2},
+  { cursor: 1, length: 2 },
+  { cursor: 3, length: 3 },
 ];
 
 export default new Vuex.Store({
   state: {
     asm: {
       activators: 1,
-      // cycles: new Array<ICycle>(),
+      nonActivators: 1,
       cycles: startingCycles,
       iteration: 0,
     },
   },
   getters: {
     anyActive(state): boolean {
-      return _.some(state.asm.cycles, (cycle) => {
-        return cycle.cursor <= state.asm.activators;
+      var isActive = false;
+      state.asm.cycles.forEach((cycle) => {
+        if (cycle.cursor <= state.asm.activators) {
+          isActive = true;
+        }
       });
+      return isActive;
     },
   },
   mutations: {
     setActivators(state, payload: number) {
-      state.asm.activators = payload;
+      state.asm.activators = Number(payload);
+      const cycleLength = state.asm.activators + state.asm.nonActivators;
+      state.asm.cycles = [
+        { cursor: 1, length: cycleLength },
+        { cursor: cycleLength + 1, length: cycleLength + 1 },
+      ];
+      state.asm.iteration = 0;
+    },
+    setNonActivators(state, payload: number) {
+      state.asm.nonActivators = Number(payload);
+      const cycleLength = state.asm.activators + state.asm.nonActivators;
+      state.asm.cycles = [
+        { cursor: 1, length: cycleLength },
+        { cursor: cycleLength + 1, length: cycleLength + 1 },
+      ];
+      state.asm.iteration = 0;
     },
     incrementIteration(state) {
       state.asm.iteration++;
+      const length = state.asm.iteration + state.asm.activators + state.asm.nonActivators + 1;
+      state.asm.cycles[state.asm.cycles.length - 1].length = 0;
+      state.asm.cycles[state.asm.cycles.length - 1].cursor = length;
+      state.asm.cycles[state.asm.cycles.length - 1].length = length;
```

Comment:
@navyhockey56 why set this to `0` first on line 60?

Reply:
@sballan I had a bug in my code originally and that had semi-fixed a problem. I just removed it and turns out it's not needed.